### PR TITLE
Transport pipeline

### LINKED ISSUE #1142: How can I use pino-socket with pino-syslog?
I have a test app:

```typescript
'use strict';

const pino = require('pino');
const transports = pino.transport({
  targets: [
    {
      target: 'pino-syslog',
    },
    {
      target: 'pino-socket',
      options: {
        address: '127.0.0.1',
        port: 5140,
        mode: 'tcp',
      },
    },
  ],
});
const logger = pino(transports);

logger.error('Test message');
```

When I run it, the syslog-formatted message logs to stdout:

```
<131>1 2021-09-28T19:15:09Z bmaupin-laptop - 218252 - - {"level":50,"time":1632856509189,"pid":218252,"hostname":"workstation","msg":"Test message"}
```

And the syslog server instead receives an unformatted log:

```
$ nc -kl 5140
{"level":50,"time":1632856509189,"pid":218252,"hostname":"workstation","msg":"Test message"}
```

These are the package versions I'm using:

```
    "pino": "^7.0.0-rc.7",
    "pino-socket": "^4.0.0",
    "pino-syslog": "github:Eomm/pino-syslog#pino-7",
```

Thanks!

### PR DESCRIPTION
**Pull request by mcollina**
Fixes #1142

### CODE REVIEW

**Comment by mcollina** [comment]
cc @jsumners @Eomm

File: lib/worker-pipeline.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        throw error
+      }
+    }
+    const stream = await fn(t.options)
+    return stream
+  }))
+  const ee = new EE()
+
+  const stream = new PassThrough({
+    destroy (err, cb) {
```
Comment (Eomm) [inline lib/worker-pipeline.js:31]:
```suggestion
    autoDestroy: true,
    destroy (err, cb) {
```

Comment (mcollina) [reply lib/worker-pipeline.js:32]: thx

**Review by jsumners — CHANGES_REQUESTED** [review]
Need doc updates.

**Comment by mcollina** [comment]
@jsumners added docs.

File: docs/transports.md [hunk 2]
```diff
@@ -143,6 +143,62 @@ module.exports = function (opts) {
...
+      target: 'pino-pretty
+    }]
+  }
+})
+
+logger.info('hello world')
+```
+
+__NOTE: there is no "default" destinations for a pipeline but
+a terminating target, i.e. a `Writable` stream.__
```
Comment (jsumners) [inline docs/transports.md:200]:
```suggestion
__NOTE: there is no "default" destination for a pipeline but
a terminating target, i.e. a `Writable` stream.__
```

Should we expose some usable destination for pipelines that dumps to `stdout`? Something like:

```js
pipeline:[
  {  /* custom transform */ },
  { target: pino.pipelineStdoutTerminator }
]
```

Comment (mcollina) [reply docs/transports.md:200]: We already have that, it's `pino/file`. By default it writes to stdout.

**Review by jsumners — APPROVED** [review]

**Pull request merged**